api/repositories.go

```diff
@@ -105,6 +105,30 @@ func (r *Repositories) Delete(name, reason string, allowDataDeletion bool) error
 	return nil
 }
 
+type DefaultGroupEnum string
+
+func (r *Repositories) AddUser(name, username, group string) error {
+	var mutation struct {
+		UpdateDefaultGroupMembershipsMutation struct {
+			ClientMutationId string
+		} `graphql:"updateDefaultGroupMemberships(input: {viewName: $name, userName: $username, groups: [$group]})"`
+	}
+	variables := map[string]interface{}{
+		"name":     graphql.String(name),
+		"username": graphql.String(username),
+		"group":    DefaultGroupEnum(group),
+	}
+
+	graphqlErr := r.client.Mutate(&mutation, variables)
+
+	if graphqlErr != nil {
+		// The graphql error message is vague if the user already is a member, so add a hint.
+		return fmt.Errorf("%+v. Does the user already have access to the repo?", graphqlErr)
```

Comment:
When I try this, adding an existing user just updates his groups without errors. Are you seeing something else?

Reply:
your right - I rewamped the old PR I had and forgot to remove this.